### ISSUE TITLE
[BUG] discriminator property of supertype not marked as so in subtype 

### ISSUE THREAD

**Issue by RAltena** [OpenAPITools/openapi-generator#11089]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x ] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?

##### Description

When I specify a type `Pet` containing a discriminator `type` and specify a subtype `Dog` using `allOf`, then the generated model  value for `Pet.type.isDiscriminator == true`  while `Dog.type.isDiscriminator == false` . I would expect  `Dog.type.isDiscriminator == true`.

This behavior is unexpected to me, but I'm not 100% sure if it's intended or not. Can someone confirm? In case this is a bug I can invest some time to try to solve it.

##### openapi-generator version
5.3.0

##### OpenAPI declaration file content or url

```yaml
    Pet:
      type: object
      discriminator:
        propertyName: type
      required: [name, type]
      properties:
        name:
          type: string
        type:
          type: string
    Dog:
      allOf:
        - $ref: "#/components/schemas/Pet"
        - type: object
          required: [name, type, age]
          properties:
            age:
              type: string
```

##### Generation Details

Using the folowing `data_class.mustache` template:
```
{{#allVars}}//{{&baseName}} isDiscriminator == {{isDiscriminator}}}
{{/allVars}}
```
You will be able to see in the generated models that for Pet, `type.isDiscriminator == true`, while for Dog the same is false. 

##### Steps to reproduce

`$ openapi-generator-cli generate -g kotlin -i openapi.yaml -o gen -t templates`

**Comment by F43nd1r** [comment]
@RAltena did you ever find a way to fix this? 

This currently blocks us from writing a template for jackson with hidden type info.

Sample generated:
```
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type", visible = false)
```
doesn't work because we can't exclude the `type` property in subtypes using `isDiscriminator`.

**Comment by RAltena** [comment]
No, I remember working around the issue, but I can not recall the details
